backend/src/org/jetbrains/demo/thinkter/Index.kt

```diff
@@ -2,36 +2,35 @@ package org.jetbrains.demo.thinkter
 
 import org.jetbrains.demo.thinkter.dao.*
 import org.jetbrains.demo.thinkter.model.*
-import org.jetbrains.ktor.application.*
-import org.jetbrains.ktor.html.*
-import org.jetbrains.ktor.http.*
-import org.jetbrains.ktor.locations.*
-import org.jetbrains.ktor.response.*
-import org.jetbrains.ktor.routing.*
-import org.jetbrains.ktor.sessions.*
+import io.ktor.application.*
+import io.ktor.html.*
+import io.ktor.http.*
+import io.ktor.locations.*
+import io.ktor.request.contentType
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.sessions.*
 import java.time.*
 
 fun Route.index(storage: ThinkterStorage) {
-    contentType(ContentType.Text.Html) {
-        get<Index> {
-            call.respondHtmlTemplate(ApplicationPage()) {
-                caption { +"Thinkter" }
-            }
-        }
-    }
-    contentType(ContentType.Application.Json) {
-        get<Index> {
-            val user = call.sessionOrNull<Session>()?.let { storage.user(it.userId) }
-            val top = storage.top(10).map(storage::getThought)
-            val latest = storage.latest(10).map(storage::getThought)
+    get<Index> {
+        val user = call.sessions.get<Session>()?.let { storage.user(it.userId) }
+        val top = storage.top(10).map(storage::getThought)
+        val latest = storage.latest(10).map(storage::getThought)
 
-            call.response.pipeline.intercept(ApplicationResponsePipeline.After) {
+        if(call.request.contentType().match(ContentType.Application.Json)) {
```

Comment:
Request's content type for GET? I believe it makes no sense as GET requests has no body. 
ETag should be always specified to get browser cache working

Reply:
I tried to preserve the original behaviour, but I couldn't get the example to work without these changes. I could use some help on this one.

Reply:
Most likely it doesn't work because of this change

```
-        override var headers: dynamic = json("Accept" to "application/json")
+        override var headers: dynamic = json("Content-Type" to contentType)
```

I'd say we actually need both `Accept` and `Content-Type` headers otherwise the server could reject it. Not sure what is the default `Accept` header value